File: src/main/java/com/demonwav/statcraft/magic/ProjectilesCode.java
/*
 * StatCraft Bukkit Plugin
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 * http://demonwav.com
 *
 * MIT License
 */

package com.demonwav.statcraft.magic;

public enum ProjectilesCode {

    NORMAL_ARROW((short)0),
    FLAMING_ARROW((short)1),
    ENDER_PEARL((short)2),
    UNHATCHED_EGG((short)3),
    HATCHED_EGG((short)4),
    SNOWBALL((short)5),
    FOUR_HATCHED_EGG((short)6);

    private short code;

    ProjectilesCode(short code) {
        this.code = code;
    }

    public short getCode() {
        return code;
    }

    public static ProjectilesCode fromCode(short code) {
        for (ProjectilesCode projectilesCode : values()) {
            if (code == projectilesCode.getCode())
                return projectilesCode;
        }
        return null;
    }
}

File: src/main/kotlin/com/demonwav/statcraft/magic/ProjectilesCode.kt
/*
 * StatCraft Bukkit Plugin
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 * https://www.demonwav.com
 *
 * MIT License
 */

package com.demonwav.statcraft.magic

enum class ProjectilesCode(val code: Short) {

    NORMAL_ARROW    (0.toShort()),
    FLAMING_ARROW   (1.toShort()),
    ENDER_PEARL     (2.toShort()),
    UNHATCHED_EGG   (3.toShort()),
    HATCHED_EGG     (4.toShort()),
    SNOWBALL        (5.toShort()),
    FOUR_HATCHED_EGG(6.toShort());

    companion object {
        @JvmStatic
        fun fromCode(code: Short): ProjectilesCode? {
            for (projectilesCode in values()) {
                if (code == projectilesCode.code)
                    return projectilesCode
            }
            return null
        }
    }
}
